ihatemoney/web.py

```diff
@@ -195,15 +195,18 @@ def admin():
 def authenticate(project_id=None):
     """Authentication form"""
     form = AuthenticationForm()
+
+    if not form.id.data and request.args.get("project_id"):
+        form.id.data = request.args["project_id"]
+    project_id = form.id.data
     # Try to get project_id from token first
     token = request.args.get("token")
     if token:
```

Comment:
What happens when a token is given, but no project_id? A test case would be nice.

Reply:
Then `verify_token` will use the wrong secret, and signature will fail.

The only test done currently is the one with a false token and no project_id in `test_invite()`

Reply:
Ok!